specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json

```diff
@@ -3562,6 +3562,408 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/threatIntelligence/indicators": {
```

Comment:
This URI indicates the user is updating a singleton "Microsoft.SecurityInsights/threatIntelligence" resource named "indicators".  I don't think thats what your intention is.  Please read through the contracts in https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#put-resource.  Everything after "providers/Microsoft.SecurityInsights" should come in pairs of "{resourceType}/{resourceName}" with an optional "/{actionName}" on the end if the operation is a POST action

Reply:
Our indicator does not have unique "name" for {resourceName}. We have ID. However, when a customer create a new Indicator, they will not provide an ID, our system will generate that ID. How do we handle this case ?
It's PUT btw.

Reply:
I think that is exactly the intention: Indicators is collection on the singleton threatintelligence.

Reply:
A collection of indicators on a singleton threatIntelligence resource would be "Microsoft.Security/threatIntelligence/default/indicators".  In that scenario what is the value of the threatIntelligence resource type?  Why not just have "Microsoft.Security/threatIntelligenceIndicators".

When the user does a PUT the PUT needs to be targeted at a named instance of a resource.  They should be able to provide that name.  You can have your own restrictions on what that name is but the user must be able to provide it if they are PUTing the resource.  Feel free to have a property inside that resource that exposes your internal "indicatorId".

Reply:
I see, when a user creates new resource, they must be able to provide a name. Our API needs to be in the format
PUT Microsoft.SecurityInsights/threatIntelligence/indicators/{name} , is that correct ?

Reply:
Sort of. It would have to be PUT Microsoft.SecurityInsights/threatIntelligence/default/indicators/{name}. (ARM requires alternating "type" and "name" segments as Chris explained in another comment.)

---
In reply to: [479481740](https://github.com/Azure/azure-rest-api-specs/pull/10463#discussion_r479481740) [](ancestors = 479481740)